src/plots/plot.ts

```diff
@@ -574,24 +573,38 @@ export class Plot extends Component {
 
   /**
    * Returns the {Plots.PlotEntity} nearest to the query point,
-   * or undefined if no {Plots.PlotEntity} can be found.
+   * or undefined if no {Plots.PlotEntity} can be found. The closest point is computed using
+   * the standard Euclidean distance formula. Note that this is done in pixel space.
    *
    * @param {Point} queryPoint
-   * @param {bounds} Bounds The bounding box within which to search. By default, bounds is the bounds of
+   * @param {Bounds} bounds The bounding box within which to search. By default, bounds is the bounds of
    * the chart, relative to the parent.
    * @returns {Plots.PlotEntity} The nearest PlotEntity, or undefined if no {Plots.PlotEntity} can be found.
    */
   public entityNearest(queryPoint: Point, bounds = this.bounds()): Plots.PlotEntity {
-    const nearest = this._getEntityStore().entityNearest(queryPoint, (entity: Plots.LightweightPlotEntity) => {
-      return this._entityVisibleOnPlot(entity, bounds);
+    let closestDistanceSquared = Infinity;
+    let closestPointEntity: Plots.LightweightPlotEntity;
+    this._getEntityStore().forEach((entity) => {
+      if (this._entityVisibleOnPlot(entity, bounds) !== false) {
+        const entityPoint = this._pixelPoint(entity.datum, entity.index, entity.dataset);
```

Comment:
Okay, i'd like the change how this works slightly.  Plots.LightweightPlotEntity already has a `position` prop. But if that is not a pixel position, what is it? There are no docs on the prop.

If `position` is computed incorrectly then we should fix that. If it's a separate kind of "Point" (like not in screen space), then i vote that we add another prop to the lightweight entity for `pixelPosition`. Example see xyplot's `_buildLightweightPlotEntities` method.